File: src/main/java/com/example/Nadeuri/comment/CommentEntity.java
package com.example.Nadeuri.comment;

import com.example.Nadeuri.board.BoardEntity;
import com.example.Nadeuri.member.MemberEntity;
import jakarta.persistence.*;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;
import java.util.List;

@Getter
@NoArgsConstructor
@EntityListeners(AuditingEntityListener.class)
@Table(name = "comment")
@Entity
public class CommentEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "comment_no", nullable = false)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "board_no", nullable = false)
    private BoardEntity board;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "member_no", nullable = false)
    private MemberEntity member;

    @Column(name = "content", nullable = false, columnDefinition = "TEXT")
    private String content;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "parent_comment_no")
    private CommentEntity parentComment;

    @OneToMany(mappedBy = "parentComment", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<CommentEntity> replies;

    @CreatedDate
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @LastModifiedDate
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @Builder
    public CommentEntity(Long id, BoardEntity board, MemberEntity member, String content, CommentEntity parentComment, List<CommentEntity> replies) {
        this.id = id;
        this.board = board;
        this.member = member;
        this.content = content;
        this.parentComment = parentComment;
        this.replies = replies;
    }

    // content 업데이트 메서드
    public void updateContent(String content) {
        this.content = content;
    }
}

File: src/main/java/com/example/Nadeuri/comment/CommentEntity.kt
package com.example.Nadeuri.comment

import com.example.Nadeuri.board.BoardEntity
import com.example.Nadeuri.member.MemberEntity
import jakarta.persistence.*
import lombok.Builder
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.LastModifiedDate
import org.springframework.data.jpa.domain.support.AuditingEntityListener
import java.time.LocalDateTime

@EntityListeners(AuditingEntityListener::class)
@Table(name = "comment")
@Entity
class CommentEntity @Builder constructor(
    @field:Column(name = "comment_no", nullable = false)
    @field:GeneratedValue(strategy = GenerationType.IDENTITY)
    @field:Id
    private val id: Long? = null, // id는 nullable로

    @field:JoinColumn(name = "board_no", nullable = false)
    @field:ManyToOne(fetch = FetchType.LAZY, optional = false)
    private val board: BoardEntity,

    @field:JoinColumn(name = "member_no", nullable = false)
    @field:ManyToOne(fetch = FetchType.LAZY, optional = false)
    private val member: MemberEntity,

    @field:Column(name = "content", nullable = false, columnDefinition = "TEXT")
    private var content: String,

    @field:JoinColumn(name = "parent_comment_no")
    @field:ManyToOne(fetch = FetchType.LAZY)
    private val parentComment: CommentEntity? = null, // nullable 처리

    @field:OneToMany(mappedBy = "parentComment", cascade = [CascadeType.ALL], orphanRemoval = true)
    private val replies: MutableList<CommentEntity> = mutableListOf() // 기본값으로 빈 리스트
) {
    @CreatedDate
    @Column(name = "created_at", nullable = false, updatable = false)
    private var createdAt: LocalDateTime? = null

    @LastModifiedDate
    @Column(name = "updated_at")
    private var updatedAt: LocalDateTime? = null

    // content 업데이트 메서드
    fun updateContent(content: String) {
        this.content = content
    }
}